frontend/appflowy_tauri/src/appflowy_app/components/auth/auth.hooks.ts

```diff
@@ -1,14 +1,30 @@
 import { currentUserActions } from '../../stores/reducers/current-user/slice';
 import { useAppDispatch, useAppSelector } from '../../stores/store';
+import { SignInPayloadPB, UserEventSignIn, UserProfilePB } from '../../../services/backend/events/flowy-user';
 
 export const useAuth = () => {
   const dispatch = useAppDispatch();
 
   const currentUser = useAppSelector((state) => state.currentUser);
 
+  async function login(email: string, password: string, name: string): Promise<UserProfilePB> {
```

Comment:
Would it be better if we use the Promise<Result<x,x>>

Reply:
refactored to use backend service